backend/src/Controllers/voterRolls.controller.ts

```diff
@@ -93,9 +76,62 @@ const getByVoterID = async (req: any, res: any, next: any) => {
     }
 }
 
+const getVoterAuth = async (req: any, res: any, next: any) => {
+
+    if (req.election.settings.voter_id_type==='IP Address'){
+        console.log(String(req.ip))
+        req.voter_id = String(req.ip)
+    } else if(req.election.settings.voter_id_type==='Email'){
+        req.voter_id = req.user.email
+    } else if(req.election.settings.voter_id_type==='IDs'){
+        req.voter_id = req.body.voter_id
+    }
+    try {
+        const voterRollEntry = await VoterRollModel.getByVoterID(req.election.election_id, req.voter_id)
+        if (!voterRollEntry)
+            return res.status('400').json({
+                error: "Voter Roll not found"
+            })
+        req.voterRollEntry = voterRollEntry
+    } catch (err) {
+        return res.status('400').json({
+            error: "Could not find voter roll entry"
+        })
+    }
+    console.log(req.voterRollEntry)
+    if (req.election.settings.voter_roll_type==='None'){
+        req.authorized_voter = true;
+        if (req.voterRollEntry.length==0){
+            //Adds voter to roll if they aren't currently
+            const NewVoterRoll = await VoterRollModel.submitVoterRoll(req.election.election_id, [req.voter_id],false)
+            if (!NewVoterRoll)
+                return res.status('400').json({
+                    error: "Voter Roll not found"
+                })
+            req.voterRollEntry = NewVoterRoll
+            req.has_voted = false
+            next()
+        } else{
+            req.has_voted = req.voterRollEntry.submitted
+            next()
+        }
+    } else if (req.election.settings.voter_roll_type==='Email' || req.election.settings.voter_roll_type==='IDs' ){
```

Comment:
From software dev meeting

We can probably avoid some confusion by renaming voter_roll_type to election_roll_type, but this would involve renaming databases so we'll save this for later